docs/src/content/docs/cookbook/dexes/stonfi.mdx

```diff
@@ -5,10 +5,362 @@ sidebar:
+
+        let forwardPayload = StonfiSwap{
+            otherTokenWallet: RouterJettonWallet,
+            refundAddress: myAddress(),
+            excessesAddress: myAddress(),
+            deadline: now() + 10000,
+            additionalData: SwapAdditionalData{
+                minOut: 1, // swap will fail if a receiver should receive less than minOut of tokens as a result
+                receiver: myAddress(),
+            }
+        };
+
+        send(SendParameters{
+            to: RouterProxyTonWallet,
+            value: GasSwapTonToJetton + GasSwapTonToJettonFwd + offerAmount,
+            body: ProxyTonTransfer{
+                queryId: 42,
+                tonAmount: offerAmount,
+                refundAddress: myAddress(),
+                forwardPayload: forwardPayload.toCell(),
+            }.toCell()
+        });
+    }
+}
+```
+
+## Liquidity provision
+
+More about liquidity provision [here](https://docs.ston.fi/docs/developer-section/api-reference-v2/example_lp_provide).
+
+It is possible to deposit liquidity by sending just 1 type of token, the pool will automatically perform a swap and use the resulting amount to mint lp tokens. Non-single side provision is done just with `bothPositive` flag equals `true`.
+
+Liquidity provisions have several common structures and messages. Don't forget to use them with the code examples below!
+
+```tact
+message(0x37c096df) ProvideLP {
+    otherTokenWallet: Address; // Address of the other Router token wallet
+    refundAddress: Address; // Address where refund will be sent if swap fails
+    excessesAddress: Address; // Address where TON excesses will be sent
+    deadline: Int as uint64; // Timestamp of execution deadline for this tx
+    additionalData: ProvideLPAdditionalData;
+}
+
+struct ProvideLPAdditionalData {
+    minLpOut: Int as coins; // Minimum required amount of lp tokens to receive
+    receiverAddress: Address; // Address where lp tokens will be sent
+    bothPositive: Bool = true; // Trigger liquidity deposit only if both token amounts are non-zero
+    fwdGas: Int as coins = 0; // Gas used to forward a message in transfer notification after mint if customPayload is present
+    customPayload: Cell? = null;
+}
+```
+
+Below are shown gas constants. All of them are taken from the sdk [STON.fi sdk](https://github.com/ston-fi/sdk). They are slightly differ from reality, but it is ok for now.
+
+```tact
+const SingleSideProvideLpJettonGas: Int = ton("1");
+const SingleSideProvideLpJettonGasFwd: Int = ton("0.8");
+const SingleSideProvideLpTonGas: Int = ton("0.8");
+const TonTransferGas: Int = ton("0.01");
+```
+
+### Jetton deposit
+
+
+```tact
+message(0xf8a7ea5) JettonTransfer {
+    queryId: Int as uint64;
+    amount: Int as coins;
+    destination: Address;
+    responseDestination: Address?;
+    customPayload: Cell? = null;
+    forwardTonAmount: Int as coins;
+    forwardPayload: Cell?; // hack for now, equivalent to Slice as remaining with first bit set to 1
```

Comment:
I wouldn't mention `Slice as remaining` thing here. Instead, explain that there's no way to represent `Either` tl-b type directly so we have to stick with this workaround that will only work if we want to store our forwardPayload in a reference (our case)